### PR TITLE
Add CI docker image for rocm builds

### PR DESCRIPTION
**Pull request by bddppq**

### CODE REVIEW

File: docker/caffe2/jenkins/build.sh [hunk 1]
```diff
@@ -34,6 +34,11 @@ if [[ "$image" == *cuda* ]]; then
   DOCKERFILE="${OS}-cuda/Dockerfile"
 fi
 
+if [[ "$image" == *rocm* ]]; then
+  ROCM_VERSION="$(echo "${image}" | perl -n -e'/rocm(\d+\.\d+.\d+)/ && print $1')"
```
Comment (bddppq) [inline docker/caffe2/jenkins/build.sh]: current latest version is 1.7.1

**Comment by pietern** [comment]
@bddppq Can you kick off an image build off of this one (after adding the jobs), just to test it can complete?

**Comment by bddppq** [comment]
@pietern Thanks for adding the new repos. These rocm docker images have now been built successfully here https://ci.pytorch.org/jenkins/job/caffe2-docker/job/py3.6-clang3.8-rocm1.7.1-ubuntu16.04/2/console.

**Review by pietern — APPROVED** [review]
LGTM

**Pull request merged**